lib/client.c

```diff
@@ -361,10 +360,11 @@ static sf_bool STDCALL log_init(const char *log_path, SF_LOG_LEVEL log_level) {
     const char *sf_log_path;
     size_t log_path_size = 1; //Start with 1 to include null terminator
     log_path_size += strlen(time_str);
-    if (log_path) {
+
+    /* The environment variables takes precedence over the specified path */
+    sf_log_path = sf_getenv("SNOWFLAKE_LOG_PATH");
+    if (sf_log_path == NULL && log_path) {
```

Comment:
The environment variable SNOWFLAKE_LOG_PATH takes precedence over the parameter so that the user can control the log directory.